gui/IrodsBrowser.py

```diff
@@ -273,11 +273,14 @@ def _fill_preview_tab(self, obj_path):
 
     def _get_object_path_name(self, row):
         """"""
-        # if self.collTable.item(row, 0).text() != '':
-        #     obj_path = self.collTable.item(row, 0).text()
-        # else:
-        obj_path = self.inputPath.text()
-        obj_name = self.collTable.item(row, 1).text()
+        if self.collTable.item(row, 1).text().startswith("/"+self.ic._session.zone):
+            print(self.collTable.item(row, 1).text())
+            sub_paths = self.collTable.item(row, 1).text().strip("/").split("/")
+            obj_path = "/"+"/".join(sub_paths[:len(sub_paths)-1])
+            obj_name = sub_paths[-1]
+        else:
```

Comment:
@chStaiger Similarly to the previous version, this doesn't actually do anything.  Column 0 was never populated from what I can tell, and now it holds the effective status.  Column 1 has only ever had the name of the data object or the collection; no `/zone`.  I recommend just leaving the simple:

```python
         obj_path = self.inputPath.text()
         obj_name = self.collTable.item(row, 1).text()
```

Does it not work properly like that?

Reply:
It works when the table is populated by browsing through collections but not when the table is filled with the search results. Search results can come from all possible paths in iRODS, so the inputPath does not provide the information needed to build the full path in iRODS. 

The problem is, that the table is used for two different purposes. Loading search results and browsing through data. The latter one would also like to do with search results next to adjusting metadata and ACLs of search results. Hence I reused the browser tab for that purpose.

I would suggest, we leave it like this for now and have a look later at the use case for searching through data and manipulating the results of the search?!

Reply:
Exercising the search, I see that my statement above is not entirely true.  Unfortunately, now the browser table is no longer consistent.  The status is replaced with 'Search'.  There is no longer a name in column index 1, but a full path.  And the size and dates columns are erased and retained from the previous situation, respectively.

Fundamentally, putting path-based results in name-based table is not going to work.  One possibility would be to add the path to the browser table to accommodate the search results, and update the search dialog with the same columns as the browser, at least in the background so they can be properly populated.

The only remaining problem is the `iRODS path:` field.  Its existence is at odds with the search results that can include multiple and other iRODS paths.  I would hate to duplicate the browser tab functionality just for the search, and removing the `iRODS path:` field is not an option.

So, what to do . . .

Reply:
I had exactly the same feeling. Concept -wise it should be a new tab, but then we are reusing so many features from the browser that this looks like replicating too much.

If we just allow to load one item from the search into the browser, then we can set the `iRODS path` field and fill columns for that item correctly. But then we are limited to one item. And this would again clash with the multi-select of the search table ...

I think we need a brainstorm for that to see which use cases we need to support with the search.

Reply:
This is an interesting one, a redesign is probably the best solution. We can find a neat to implement this, but I agree with Christine that its probably best to think this through as a separate feature.

Reply:
Indeed, I believe it is best to leave it as is for now.  I only just found where to initiate a search from ;p  We can merge this as is and open an Issue/discussion to iron out the details.  The only question is when and where?